components/search/search.tsx

```diff
@@ -70,9 +65,13 @@ export const SearchWithIcon = ({
       return;
     }
 
-    const { search_id, imgURL } = searchResponse;
+    const { search_id, imgURL, results } = searchResponse;
 
     localStorage.setItem(`image_url_${search_id}`, imgURL ?? "");
+    localStorage.setItem(
```

Comment:
@Anubhav-Goyal01 I'd suggest using an abstraction for the localStorage stuff so that we don't need to repeat the key names everywhere and have a central place to manage things. 

Check this out: 
- https://github.com/AdiPat/ox/blob/main/client/src/common/local-storage-cache.ts
- https://github.com/AdiPat/ox/blob/main/client/src/common/cache.ts

Reply:
Makes sense.